Sources/CacheAdvance/CacheAdvance.swift

```diff
@@ -189,7 +189,7 @@ public final class CacheAdvance<T: Codable> {
     /// - Parameter messageLength: the length of the next message that will be written.
     private func prepareReaderForWriting(dataOfLength messageLength: Bytes) throws {
         while writer.offsetInFile < reader.offsetInFile
```

Comment:
I wonder if we can explain why each of these checks is important. I'm having trouble reminding myself of why we need the first one. Maybe something about how we can overflow back to the start of the file?

Also I wonder if the comment inside the while loop doesn't fully capture both situations.

Reply:
I'll make this more clear on a line-by-line basis. Good call-out.